skada/metrics.py

```diff
@@ -381,22 +381,21 @@ def _score(self, estimator, X, y, sample_domain=None, **kwargs):
                 f"The estimator {estimator!r} does not."
             )
 
+        has_transform_method = False
+
         if not isinstance(estimator, BaseEstimator):
             # The estimator is a deep model
             transformer = estimator.predict_features
+            has_transform_method = True
         else:
             # We need to find the last layer of the pipeline with a transform method
             pipeline_steps = list(enumerate(estimator.named_steps.items()))
 
             for index_transformer, (_, step_process) in reversed(pipeline_steps):
                 if hasattr(step_process, "transform"):
                     transformer = estimator[: index_transformer + 1].transform
+                    has_transform_method = True
                     break  # Stop after the first occurrence if there are multiple
-            else:
```

Comment:
Maybe it could be better to define identity function:
`def identity(x): return x` and if there is no transform method, you can put `transformer = identity`. It will be clearer I think